### PR TITLE
Generate Gdk classes

### LINKED ISSUE #134: Failure building gdk on Rust 1.13
My projects have been targetting Rust 1.13 as a minimum build version and I recently ran into a problem building gdk (through the glib and gtk dependencies) for my GUI program:
```
     Running `rustc /root/.cargo/git/checkouts/gdk-0e2106e2e12a0cc1/8b3342528ee9c85bf5ecf97c788b2cc3d531e7d8/src/lib.rs --crate-name gdk --crate-type lib -g --cfg feature=\"v3_6\" --cfg feature=\"pango\" --cfg feature=\"v3_16\" --cfg feature=\"v3_12\" --cfg feature=\"v3_14\" --cfg feature=\"v3_8\" --cfg feature=\"gdk-pixbuf\" --cfg feature=\"v3_18\" --cfg feature=\"v3_10\" --cfg feature=\"gio\" --cfg feature=\"v3_20\" --cfg feature=\"cairo-rs\" --cfg feature=\"gdk-sys\" -C metadata=1f87422b5add9ed8 -C extra-filename=-1f87422b5add9ed8 --out-dir /builds/susurrus/gattii/target/debug/deps --emit=dep-info,link -L dependency=/builds/susurrus/gattii/target/debug/deps --extern bitflags=/builds/susurrus/gattii/target/debug/deps/libbitflags-1efefdc447eb53f2.rlib --extern glib_sys=/builds/susurrus/gattii/target/debug/deps/libglib_sys-8f1dec70df9c8640.rlib --extern cairo=/builds/susurrus/gattii/target/debug/deps/libcairo-77d84e4bc57ccadc.rlib --extern pango=/builds/susurrus/gattii/target/debug/deps/libpango-7d1db9e45a06c17d.rlib --extern gdk_sys=/builds/susurrus/gattii/target/debug/deps/libgdk_sys-cd0070267421ce7b.rlib --extern libc=/builds/susurrus/gattii/target/debug/deps/liblibc-e1db4c5f3a4f3c2f.rlib --extern gdk_pixbuf=/builds/susurrus/gattii/target/debug/deps/libgdk_pixbuf-06be090bf3460918.rlib --extern gio=/builds/susurrus/gattii/target/debug/deps/libgio-f3f1e1d7dd69eae8.rlib --extern glib=/builds/susurrus/gattii/target/debug/deps/libglib-05904638e40d7bdc.rlib --cap-lints allow -L native=/usr/lib/x86_64-linux-gnu -L native=/usr/lib/x86_64-linux-gnu -L native=/usr/lib/x86_64-linux-gnu -L native=/usr/lib/x86_64-linux-gnu -L native=/usr/lib/x86_64-linux-gnu -L native=/usr/lib/x86_64-linux-gnu -L native=/usr/lib/x86_64-linux-gnu`
error[E0252]: a type named `EventMask` has already been imported in this module
   --> /root/.cargo/git/checkouts/gdk-0e2106e2e12a0cc1/8b3342528ee9c85bf5ecf97c788b2cc3d531e7d8/src/lib.rs:136:9
    |
24  | pub use auto::*;
    |         -------- previous import of `EventMask` here
...
136 | pub use ffi::GdkEventMask as EventMask;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ already imported

error[E0252]: a type named `WindowHints` has already been imported in this module
   --> /root/.cargo/git/checkouts/gdk-0e2106e2e12a0cc1/8b3342528ee9c85bf5ecf97c788b2cc3d531e7d8/src/lib.rs:151:9
    |
24  | pub use auto::*;
    |         -------- previous import of `WindowHints` here
...
151 | pub use ffi::GdkWindowHints as WindowHints;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ already imported

error[E0252]: a type named `DragAction` has already been imported in this module
   --> /root/.cargo/git/checkouts/gdk-0e2106e2e12a0cc1/8b3342528ee9c85bf5ecf97c788b2cc3d531e7d8/src/lib.rs:134:9
    |
24  | pub use auto::*;
    |         -------- previous import of `DragAction` here
...
134 | pub use ffi::GdkDragAction as DragAction;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ already imported

error: aborting due to 3 previous errors

error: Could not compile `gdk`.
```

These failures occur on `x86_64-unknown-linux-gnu`, `x86_64-pc-windows-gnu`, and `i686-pc-windows-gnu` only for 1.13. The same codebase for nightly and stable (currently 1.15.1) builds succeed.

I guess my first question is this an actual bug, either with our code of 1.13. And secondly, what is the oldest Rust supported by gtk-rs and friends as the README doesn't specify anything?

### LINKED ISSUE #143: Generate classes that currently manual
Seems most gdk classes that currently done manual way can be generated now, it add signals support.
Also better generate Monitor and possibly Sink to have full Gdk screen management.

### PR DESCRIPTION
**Pull request by EPashkin**
Fixes #143
Fixes #134

cc @GuillaumeGomez

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Yeay! \o/

**Comment by EPashkin** [comment]
Maybe also closes #134

**Comment by GuillaumeGomez** [comment]
@EPashkin: I updated your comment. Merging now, thanks!

File: Gir.toml [hunk 1]
```diff
@@ -9,27 +9,52 @@ generate_safety_asserts = true
...
+    "Gdk.WMDecoration",
+    "Gdk.WMFunction",
 ]
 
 manual = [
-    "Gdk.AppLaunchContext",
     "Gdk.Atom",
-    "Gdk.Cursor",
-    "Gdk.DeviceManager",
-    "Gdk.Device",
```
Comment (Susurrus) [inline Gir.toml]: What happened to `Device`? It's not under `generate`.

Comment (Susurrus) [inline Gir.toml]: Nvm, sorry about the noise.

**Pull request merged**